### DISCUSSION TITLE
How i can use Github

### DISCUSSION THREAD

**Discussion by samzazz** [community/community#64161] [New to GitHub]
Hello everyone and I am delighted to join your community with pride........ I am confused and do not have that amount of programming experience, I hope you give me or tell me some (main) methods or procedures that enable me to use the program to use it on Windows 11, and thank you very much .

**Comment by jadstrike** [comment]
Firstly, install git on your windows and then do these step by step.
**Step 1**
Create a simple project using any tech stack u like eg.(plain html css js, react,..)
**Step 2**
Learn version control using that project
here are the some resources that I like
[https://youtu.be/tRZGeaHPoaw](url)
[https://youtu.be/mJ-qvsxPHpY](url)
**Step 3**
make a repo on github and practice pushing to the origin main and then you can see your project on github. Follow the instructions and watch the tutorials along the way. You will be fine for sure. Just give this steps a try for 2 days and you will get the foundation of git and github.

**Comment by gamal2001** [comment]
GitHub is a platform that allows you to host, manage, and collaborate on Git repositories. You can use GitHub to create and contribute to open-source projects, work on personal or professional projects, and connect with other developers. Here are some of the basic steps to use GitHub:


- **Set up Git**: Git is the version control system that GitHub uses to track changes and the history of your code. You need to install Git on your computer and configure it with your name and email address. You can also use GitHub CLI, GitHub Desktop, or GitHub Mobile to interact with Git².
- **Create a repository**: A repository is where you store your code and related files. You can create a new repository on GitHub or import an existing one from another source. You can also fork a repository from another user to create a copy of it on your account².
- **Create branches**: Branches are different versions of your repository that you can use to work on different features or experiments. You can create new branches from the main branch or from other branches. You can also merge branches to combine your changes².
- **GitHub flow**: GitHub flow is a workflow that helps you collaborate on projects using branches, pull requests, and reviews. You can use GitHub flow to propose changes, get feedback, and deploy your code².
- **Contributing to projects**: You can contribute to other projects on GitHub by creating issues, commenting on discussions, or submitting pull requests. You can also join organizations or teams to work with other people on shared projects².
- **Be social**: GitHub is also a social network for developers. You can follow other users, star or watch repositories, join communities, and participate in events. You can also communicate with other users using comments, mentions, reactions, or emojis².



here are some links that will help you to learn more about GitHub

(1) Get started with GitHub documentation - GitHub Docs. https://docs.github.com/en/get-started.
([2](https://docs.github.com/en/get-started/quickstart)) Quickstart - GitHub Docs. https://docs.github.com/en/get-started/quickstart.
(3) Get started with [GitHub documentation](https://docs.github.com/en/get-started) - GitHub Docs. https://docs.github.com/en/get-started.
(4) How To Use GitHub | GitHub Tutorial For Beginners | Edureka. https://www.edureka.co/blog/how-to-use-github/.
(5) Quickstart - GitHub Docs. https://docs.github.com/en/get-started/quickstart.

**Comment by LeandroLuna** [comment]
Hi, samzazz!

The easiest and quickest method to begin using GitHub is through [GitHub Desktop](https://desktop.github.com/). It offers a user-friendly and intuitive interface for effortlessly creating and managing your projects.
